process/peer/peerTypeProvider.go

```diff
@@ -62,12 +68,12 @@ func NewPeerTypeProvider(arg ArgPeerTypeProvider) (*PeerTypeProvider, error) {
 		validatorsProvider:           arg.ValidatorsProvider,
 		cache:                        make(map[string]*peerListAndShard),
 		mutCache:                     sync.RWMutex{},
-		cacheRefreshIntervalDuration: arg.CacheRefreshIntervalDuration,
+		cacheRefreshIntervalDuration: arg.PeerTypeRefreshIntervalInSec,
 		refreshCache:                 make(chan bool),
 	}
 
-	ptp.startRefreshProcess()
-
+	currentContext, _ := context.WithCancel(arg.Context)
```

Comment:
do we need to use the passed context? 
can't we use the conotext.background ?

Reply:
made the peerType provider a CLoser and used context.Background